tests/h/services/document_test.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import pytest
+
+from h.services.document import DocumentService
+
+
+class TestFetchByGroupid(object):
+    def test_it_returns_documents_annotated_within_group(
+        self, svc, groups, annotations, matchers
+    ):
+        docs = svc.fetch_by_groupid(groupid=groups["target_group"].pubid)
+
+        assert docs == matchers.UnorderedList([anno.document for anno in annotations])
+
+    def test_it_does_not_return_documents_annotated_in_other_groups(
+        self, groups, annotations, other_annotations, svc, matchers
+    ):
+        docs = svc.fetch_by_groupid(groupid=groups["target_group"].pubid)
+        for other_anno in other_annotations:
+            assert other_anno.document not in docs
+
+    def test_it_returns_each_document_only_once(
+        self, svc, groups, annotations, db_session
+    ):
+        # This will make both annotations point to the same document
+        annotations[1].document = annotations[0].document
+        annotations[2].document = annotations[0].document
+        db_session.commit()
```

Comment:
You should always use `flush()` rather than `commit()` because it's faster (and in production code also less destructive), unless you really have to `commit()`, but I don't think you ever have to. In this specific test file it looks like no `commit()`s or `flush()`s are necessary -- you can just delete them and it still works. The reason is probably because sqlalchemy _queries_ cause a `flush()` to happen automatically